File: VSCode_CPP/56.merge-intervals.cpp
/*
 * @lc app=leetcode id=56 lang=cpp
 *
 * [56] Merge Intervals
 */

// @lc code=start
class Solution {
   public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> merged;
        sort(intervals.begin(), intervals.end());
        for (int i = 0; i < intervals.size(); i++) {
            int L = intervals[i][0], R = intervals[i][1];
            if (!merged.empty() && merged.back()[1] >= L)
                merged.back()[1] = max(R, merged.back()[1]);
            else
                merged.push_back(intervals[i]);
        }
        return merged;
    }
};
// @lc code=end

File: Interview/2023/56.merge-intervals.py
#
# @lc app=leetcode id=56 lang=python3
#
# [56] Merge Intervals
#

# @lc code=start
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        s = []
        for interval in sorted(intervals, key=lambda x:x[0]):
            if s and interval[0] <= s[-1][1]:
                s[-1][1] = max(s[-1][1], interval[1])
            else:
                s.append(interval)
        return s

# @lc code=end

